data-prepper-plugin-framework/src/test/java/org/opensearch/dataprepper/plugin/PluginBeanFactoryProviderTest.java

```diff
@@ -112,8 +113,32 @@ void testCreatePluginSpecificContext() {
     void testCreatePluginSpecificContext_with_empty_array() {
         when(context.getParent()).thenReturn(context);
         final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
-        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{});
+        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{}, null);
         assertThat(beanFactory, notNullValue());
         assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestComponent.class));
     }
+
+    @Test
+    void testCreatePluginSpecificContext_with_empty_array_with_plugin_config() {
+        when(context.getParent()).thenReturn(context);
+        final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
+        TestPluginConfiguration config = new TestPluginConfiguration();
+        BeanFactory beanFactory = objectUnderTest.createPluginSpecificContext(new Class[]{}, config);
+        assertThat(beanFactory, notNullValue());
+        assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestComponent.class));
+        assertThrows(NoSuchBeanDefinitionException.class, ()->beanFactory.getBean(TestPluginConfiguration.class));
+    }
+
+    @Test
+    void testCreatePluginSpecificContext_with_plugin_config() {
+        when(context.getParent()).thenReturn(context);
+        final PluginBeanFactoryProvider objectUnderTest = createObjectUnderTest();
+        TestPluginConfiguration config = new TestPluginConfiguration();
+        config.setRequiredString("required_string_value");
```

Comment:
It would be better to set this to a random value and then verify against that.

```
String requiredStringValue = UUID.randomUUID().toString();
config.setRequiredString(requiredStringValue);

...
assertThat(beanFactory.getBean(TestPluginConfiguration.class).getRequiredString(), equalTo(requiredStringValue));
```

Reply:
Modified to use a random string